Cabal-syntax/src/Distribution/Types/GenericPackageDescription.hs

```diff
@@ -86,19 +85,24 @@ instance L.HasBuildInfos GenericPackageDescription where
             <*> (traverse . L._2 . traverseCondTreeBuildInfo) f x4
             <*> (traverse . L._2 . traverseCondTreeBuildInfo) f x5
             <*> (traverse . L._2 . traverseCondTreeBuildInfo) f x6
-      where
 
 -- We use this traversal to keep [Dependency] field in CondTree up to date.
 traverseCondTreeBuildInfo
     :: forall f comp v. (Applicative f, L.HasBuildInfo comp)
     => LensLike' f (CondTree v [Dependency] comp) L.BuildInfo
 traverseCondTreeBuildInfo g = node where
-    mkCondNode :: comp -> [CondBranch v [Dependency] comp] -> CondTree v [Dependency] comp
-    mkCondNode comp branches = CondNode comp (view L.targetBuildDepends comp) branches
+    mkCondNode
+      :: comp
+      -> [Dependency]
+      -> [CondBranch v [Dependency] comp]
+      -> CondTree v [Dependency] comp
+    mkCondNode comp deps = CondNode comp (view L.targetBuildDepends comp <> deps)
```

Comment:
This naive concatenation makes the test suite fail, as I suspected. I do think the issue is with this field, but it may be due to its misuse before the call to `traverseCondTreeBuildInfo`.